src/peft/peft_model.py

```diff
@@ -386,20 +395,36 @@ def load_adapter(self, model_id, adapter_name, is_trainable=False, **kwargs):
         # load weights if any
         path = os.path.join(model_id, kwargs["subfolder"]) if kwargs.get("subfolder", None) is not None else model_id
 
-        if os.path.exists(os.path.join(path, WEIGHTS_NAME)):
+        if os.path.exists(os.path.join(path, SAFETENSORS_WEIGHTS_NAME)):
+            filename = os.path.join(path, SAFETENSORS_WEIGHTS_NAME)
+            use_safetensors = True
+        elif os.path.exists(os.path.join(path, WEIGHTS_NAME)):
             filename = os.path.join(path, WEIGHTS_NAME)
+            use_safetensors = False
         else:
             try:
-                filename = hf_hub_download(model_id, WEIGHTS_NAME, subfolder=kwargs.get("subfolder", None), **kwargs)
-            except:  # noqa
-                raise ValueError(
-                    f"Can't find weights for {model_id} in {model_id} or in the Hugging Face Hub. "
-                    f"Please check that the file {WEIGHTS_NAME} is present at {model_id}."
+                # Priority 1: load safetensors weights
+                filename = hf_hub_download(
+                    model_id, SAFETENSORS_WEIGHTS_NAME, subfolder=kwargs.get("subfolder", None), **kwargs
                 )
+            except:  # noqa
+                try:
+                    filename = hf_hub_download(
+                        model_id, WEIGHTS_NAME, subfolder=kwargs.get("subfolder", None), **kwargs
+                    )
+                except:  # noqa
+                    raise ValueError(
+                        f"Can't find weights for {model_id} in {model_id} or in the Hugging Face Hub. "
+                        f"Please check that the file {WEIGHTS_NAME} or {SAFETENSORS_WEIGHTS_NAME} is present at {model_id}."
+                    )
```

Comment:
I don't like the double try / except will find a better workaround

Reply:
Will use `cached_file` method from `transformers`: https://github.com/huggingface/transformers/blob/02fe3af275f0ad935441b813ef2e58b94d09788a/src/transformers/utils/hub.py#L300